### DISCUSSION TITLE
How to handle closing of a dialog?

### DISCUSSION THREAD

**Discussion by vopani** [h2oai/wave#789] [Q&A] [answered]
I have a dialog box with `closable=True`.

What exactly happens when the close button is clicked? I would expect the dialog to not show up again. But it does (when something else is triggered).
Ok, so then I added `q.page['meta'].dialog = None` whenever the serving function is called. That does remove it and it works. But the dialog box reappears for a brief moment before it vanishing. I believe it's because when the serving function is called, the dialog box is still present in meta and takes a moment for `= None` to get executed.

Any way around this?

Even a bare minimal example to show how `closable=True` is to be used will help. Otherwise `closable` is not usable.

**Comment by lo5** [comment, marked as answer]
> What exactly happens when the close button is clicked? I would expect the dialog to not show up again. 

Yes, this is a bug. It should not show up again. 
Clicking the close button should set the dialog to `None` without you having to do `q.page['meta'].dialog = None`.

> Otherwise closable is not usable.

Agree.

**Reply by lo5** [reply]
Tracked here: https://github.com/h2oai/wave/issues/790

**Reply by vopani** [reply]
Yes, that would be perfect. Thanks!
